hash/hash.go

```diff
@@ -124,29 +106,15 @@ func (h Hash) Size() int {
 	if len(h) == 0 {
 		return 0
 	}
-	return base58size
+	return size
 }
 
-// MarshalJSON mashals hash into base58 encoded json string.
+// MarshalJSON mashals hash into encoded json string.
 func (h Hash) MarshalJSON() ([]byte, error) {
-	return json.Marshal(base58.Encode(h))
+	return json.Marshal([]byte(h))
```

Comment:
I suggest that the JSON version to be a base58 encoded string.
I the current implementation will put the hash in a binary form inside the json...

Reply:
It will display it as base64: https://play.golang.org/p/q59QB-dZutT
```go
s := S{Bin: []byte("hello world")}
d, _ := json.Marshal(s)
fmt.Println(string(d))
```
produces:
```
{"Bin":"aGVsbG8gd29ybGQ="}
```